jupyterhub/auth.py

```diff
@@ -125,6 +125,15 @@ def validate_username(self, username):
         """
     ).tag(config=True)
 
+    delete_invalid_users = Bool(False,
+        help="""Delete any invalid users from the database
```

Comment:
Perhaps:

```
help="""Delete users from the hub database if users fail validation

When the JupyterHub server starts, a `validate_users` check will be performed
on the Hub's database. 

If set to True, any users that do not pass validation will be
deleted.

If set to False (default), invalid users remain in the Hub's database. It is the
default to avoid data loss due to config changes.
```